File: mirai-core/src/main/java/net/mamoe/mirai/message/PlainText.java
package net.mamoe.mirai.message;

/**
 * @author Him188moe
 */
public final class PlainText extends Message {
    private final String text;

    public PlainText(String text) {
        this.text = text;
    }

    @Override
    public String toString() {
        return text;
    }
}

File: mirai-core/src/commonMain/kotlin/net.mamoe.mirai/message/PlainText.kt
package net.mamoe.mirai.message


inline class PlainText(override val stringValue: String) : Message {
    override operator fun contains(sub: String): Boolean = sub in stringValue
    override fun toString(): String = stringValue

    companion object Key : Message.Key<PlainText>
}

/**
 * 构造 [PlainText]
 */
fun String.toMessage(): PlainText = PlainText(this)

/**
 * 得到包含作为 [PlainText] 的 [this] 的 [MessageChain].
 *
 * @return 唯一成员且不可修改的 [SingleMessageChainImpl]
 *
 * @see SingleMessageChain
 * @see SingleMessageChainImpl
 */
fun String.singleChain(): MessageChain = this.toMessage().chain()